Auth nav unit tests

zhen0 opened the pull request:
PR Checklist:

- [x] add a short description of what's changed to the top of the `CHANGELOG.md`
- [x] add/update tests (or don't, for reasons explained below)

## Describe this PR
- Adds unit tests for the auth nav guard middleware
- Closes #150

znicholasbrown commented on tests/unit/middleware/authNavGuard.spec.js:
```diff
@@ -0,0 +1,205 @@
+    await authNavGuard({}, {}, next)
+
+    expect(store.getters['auth0/isAuthorized']).toBe(true)
+
```
Same comment as above re: adding an assertion that `auth0/authorize` is called

znicholasbrown commented on tests/unit/middleware/authNavGuard.spec.js:
```diff
@@ -0,0 +1,205 @@
+    await authNavGuard({}, {}, next)
+
+    expect(store.getters['user/userIsSet']).toBe(true)
+
```
Same comment as above re: adding an assertion that `user/getUser` is called

znicholasbrown reviewed (dismissed):
Just a few comments but all the same tweak @zhen0

zhen0 commented:
Thanks for the comments @znicholasbrown - I haven't added a check for 'authenticate' or 'authorize' being called because I could not get spying on store.dispatch or store.actions to work.  I've updated the wording of the tests and added comments to reflect what we are testing and also added an issue in #297 to come back and to figure out that spy. 

I've also updated the tenant store unit tests to remove a mockError that was causing a pre-commit hook error for @whimsicallyson - standard git commit should work for now.

znicholasbrown reviewed (approved):
Approved with the new changes 👍
